psydac/ddm/cart.py

```diff
@@ -841,47 +874,50 @@ def _compute_shift_info( self, direction, disp ):
+
         if disp > 0:
-            recv_starts[direction] = 0
-            send_starts[direction] = e-s+1
+            recv_starts[direction]          = 0
+            send_starts[direction]          = e-s+1
+            recv_assembly_starts[direction] = 0
+            send_assembly_starts[direction] = e-s+1+m*p
         elif disp < 0:
-            recv_starts[direction] = e-s+1+m*p
-            send_starts[direction] = m*p
+            recv_starts[direction]          = e-s+1+m*p
+            send_starts[direction]          = m*p
+            recv_assembly_starts[direction] = e-s+1+m*p
+            send_assembly_starts[direction] = 0
 
         # Store all information into dictionary
-        info = {'rank_dest'  : rank_dest,
-                'rank_source': rank_source,
-                'buf_shape'  : tuple(  buf_shape  ),
-                'send_starts': tuple( send_starts ),
-                'recv_starts': tuple( recv_starts )}
+        info = {'rank_dest'           : rank_dest,
+                'rank_source'         : rank_source,
+                'buf_shape'           : tuple(  buf_shape  ),
+                'send_starts'         : tuple( send_starts ),
+                'recv_starts'         : tuple( recv_starts ),
+                'send_assembly_starts': tuple( send_assembly_starts ),
+                'recv_assembly_starts': tuple( recv_assembly_starts )}
         return info
 
 #===============================================================================
-class InterfaceCartDecomposition(CartDecomposition):
+class InterfaceCartDecomposition:
     """
-    The Cartesian decomposition of an interface constructed from the Cartesian decomposition of the patches that share an interface.
-    This is built using a new inter-communicator between the cartesian grids.
+    The Cartesian decomposition of an interface constucted from the Cartesian decomposition of the patches that shares an interface.
+    This is built using a new inter-communicator between the cartition grids.
 
     Parameters
     ----------
-    npts : list
-        Number of coefficients in the global grid along each dimension for the patches that shares the interface.
 
-    pads : list
-        Padding along each grid dimension for the patches that shares the interface.
+    cart_minus: CartDecomposition
+        The cartesian decomposition of the minus patch.
 
-    periods : list or tuple of bool
-        Periodicity (True|False) along each grid dimension for the patches that shares the interface.
+    cart_plus: CartDecomposition
+        The cartesian decomposition of the plus patch.
 
     comm : mpi4py.MPI.Comm
         MPI communicator that will be used to spawn the cart decomposition
 
-    shifts: list or tuple of int
-        Shifts along each grid dimension.
-        It takes values bigger or equal to one, it represents the multiplicity of each knot.
-
     axes: list of ints
         The axes of the patches that constucts the interface.
```

Comment:
Typo: constucts -> construct

But here I suggest the word "share" instead

Reply:
fixed in [a07719b](https://github.com/pyccel/psydac/pull/232/commits/a07719bbf5eb7bfe9d65e8262d174d97ba239c10)